scripts/abins/input/dmol3loader.py

```diff
@@ -41,245 +55,233 @@ def read_vibrational_or_phonon_data(self):
+                unit_cell *= ATOMIC_LENGTH_2_ANGSTROM
 
-    def _convert_to_angstroms(self, string=None):
-        """
-        :param string: string with number
-        :returns: converted coordinate of lattice vector to Angstroms
-        """
-        au2ang = ATOMIC_LENGTH_2_ANGSTROM
-        return float(string) * au2ang
+        return unit_cell
 
-    def _read_atomic_coordinates(self, file_obj=None, data=None, masses_from_file=None):
+    @staticmethod
+    def _read_atomic_coordinates(file_obj: io.BufferedReader) -> dict:
         """
         Reads atomic coordinates from .outmol DMOL3 file.
 
-        :param file_obj: file object from which we read
-        :param data: Python dictionary to which atoms data should be added
-        :param masses_from_file: masses read from an ab initio output file
+        :param file_obj: open file handle to DMOL3 output data
         """
         atoms = {}
-        atom_indx = 0
-        self._parser.find_first(file_obj=file_obj, msg="$coordinates")
-        end_msgs = ["$end", "----------------------------------------------------------------------"]
-
-        while not self._parser.block_end(file_obj=file_obj, msg=end_msgs):
+        atom_index = 0
+        TextParser.find_first(file_obj=file_obj, msg="$coordinates")
+        end_msgs = ["$end", "------------------------------------------------"]
 
-            line = file_obj.readline()
-            entries = line.split()
+        while not TextParser.block_end(file_obj, msg=end_msgs):
+            symbol, *coord = file_obj.readline().split()
 
-            symbol = str(entries[0].decode("utf-8").capitalize())
+            symbol = symbol.decode("utf-8").capitalize()
             atom = Atom(symbol=symbol)
-            # We change unit of atomic displacements from atomic length units to Angstroms
-            au2ang = ATOMIC_LENGTH_2_ANGSTROM
-            float_type = FLOAT_TYPE
 
-            atoms["atom_{}".format(atom_indx)] = {"symbol": symbol, "mass": atom.mass, "sort": atom_indx,
-                                                  "coord": np.asarray(entries[1:]).astype(dtype=float_type) * au2ang}
+            # Convert coordinate units from Bohr to Angstrom
+            coord = np.asarray(coord,
+                               dtype=FLOAT_TYPE) * ATOMIC_LENGTH_2_ANGSTROM
 
-            atom_indx += 1
+            atoms[f"atom_{atom_index}"] = {"symbol": symbol, "mass": atom.mass,
+                                           "sort": atom_index, "coord": coord}
+            atom_index += 1
 
-        self.check_isotopes_substitution(atoms=atoms, masses=masses_from_file)
+        return atoms
 
-        data["atoms"] = atoms
+    @staticmethod
+    def _get_to_next_nonempty(file_obj: io.BufferedReader) -> None:
+        # Scroll to next non-empty line
+        last_line = file_obj.tell()
+        for line in file_obj:
+            if line.decode('utf8').strip():
+                file_obj.seek(last_line)
+                break
```

Comment:
could this just be a return instead of a break?

Reply:
It could, if you think that would be clearer? I'm a bit wary of `return` statements in the middle of functions but this seems like a good place to use one.

Reply:
I am always wary of breaks. I find it takes more thought to work out where the code goes to next.